### PR TITLE
Making Hash Figures Available

### PR DESCRIPTION
**Pull request by ehsteve**
Small changes so that figures which are created are more easily accessible for comparison or for committing to the sunny-figure-test repository.

### CODE REVIEW

**Comment by ehsteve** [comment]
This helps to address #1610.

**Comment by Cadair** [comment]
Looks good to me. Will this always save the figures for every test run, or just when the tests fail?

**Comment by ehsteve** [comment]
It saves for every test run but they are always stored in a temp folder. There was no other way to do this that I could think of because you can't know if the test is going to fail as you create the figure unless we wanted to do two runs of making figures.

**Comment by Cadair** [comment]
@ehsteve ah yes good point. Anyway having it done everytime is useful as you can replay back in the history, generate and then go back to a broken test and see the difference.

**Comment by ehsteve** [comment]
@ayshih ping.

File: sunpy/conftest.py [hunk 1]
```diff
@@ -50,6 +50,14 @@ def pytest_runtest_setup(item):
             pytest.skip(msg.format(item.name))
 
 def pytest_unconfigure(config):
+    tempdir = tempfile.mkdtemp(suffix="_figures")
+    for h in hash.file_list:
+        for name, search_h in hash.hash_library.iteritems():
+            if h == search_h:
```
Comment (ayshih) [inline sunpy/conftest.py]: By inspection, it doesn't seem like you need the outer `for` loop.  You could just try to access `file_list` using the hash (e.g., `hash.file_list[search_h]`) and bail out if it fails.

**Comment by ayshih** [comment]
> It saves for every test run but they are always stored in a temp folder. There was no other way to do this that I could think of because you can't know if the test is going to fail as you create the figure unless we wanted to do two runs of making figures.

Instead of putting the code into `pytest_unconfigure`, you could put it into the `figure_test` decorator.  There, you could (in principle) delete the PNG if the test succeeds or move it to the temp folder if the test fails.  (The temp folder would need to be created at the configure stage, and then presumably saved in `hash` so the decorator can know where to move things.)

**Comment by ehsteve** [comment]
@ayshih I personally prefer that all of the figures be saved so that people can familiarize themselves with them and also check the ones that "should" be correct. There is very little downside to this right now. So I would advocate that we leave this for now and maybe update it later.

File: sunpy/conftest.py [hunk 2]
```diff
@@ -50,6 +50,14 @@ def pytest_runtest_setup(item):
...
+                os.rename(hash.file_list[h], os.path.join(tempdir, name + '.png'))
+                #print(os.path.join(os.path.dirname(hash.file_list[h]), name + '.png'))
```
Comment (dpshelio) [inline sunpy/conftest.py]: commented code, out! :)

File: sunpy/conftest.py [hunk 3]
```diff
@@ -50,6 +50,14 @@ def pytest_runtest_setup(item):
...
+    print('All test files for figure hashes can be found in %s' % tempdir)
```
Comment (dpshelio) [inline sunpy/conftest.py]: change it to `'...{}...'.format()` sintax, please...

**Comment by Cadair** [comment]
:+1:

**Pull request merged**